### ISSUE TITLE
Failed to compile with ffmpeg

### ISSUE THREAD

**Issue by denissabramovs** [opencv/opencv#21766]
Can't compile, ffmpeg problem.

Using latest cloned repo:
commit 9d16612a9b028f03747037c74d9678ef72e95e85 (HEAD -> 4.x, origin/master, origin/HEAD, origin/4.x)

ffmpeg:
```
ffmpeg version 4.4-6ubuntu5 Copyright (c) 2000-2021 the FFmpeg developers
built with gcc 11 (Ubuntu 11.2.0-7ubuntu1)
configuration: --prefix=/usr --extra-version=6ubuntu5 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
libavutil      56. 70.100 / 56. 70.100
libavcodec     58.134.100 / 58.134.100
libavformat    58. 76.100 / 58. 76.100
libavdevice    58. 13.100 / 58. 13.100
libavfilter     7.110.100 /  7.110.100
libswscale      5.  9.100 /  5.  9.100
libswresample   3.  9.100 /  3.  9.100
libpostproc    55.  9.100 / 55.  9.100
```

OS:
```
PRETTY_NAME="Ubuntu 21.10"
NAME="Ubuntu"
VERSION_ID="21.10"
VERSION="21.10 (Impish Indri)"
VERSION_CODENAME=impish
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=impish
```

Error log:
```
In file included from /var/local/acp/ai/opencv/modules/videoio/src/cap_ffmpeg_impl.hpp:105,
                 from /var/local/acp/ai/opencv/modules/videoio/src/cap_ffmpeg.cpp:50:
/var/local/acp/ai/opencv/modules/videoio/src/cap_ffmpeg_hw.hpp: In function ‘AVCodec* hw_find_codec(AVCodecID, AVHWDeviceType, int (*)(const AVCodec*), const char*, AVPixelFormat*)’:
/var/local/acp/ai/opencv/modules/videoio/src/cap_ffmpeg_hw.hpp:724:35: error: ‘av_codec_iterate’ was not declared in this scope; did you mean ‘av_codec_next’?
  724 |     while (NULL != (c = (AVCodec*)av_codec_iterate(&opaque)))
      |                                   ^~~~~~~~~~~~~~~~
      |                                   av_codec_next
make[2]: *** [modules/videoio/CMakeFiles/opencv_videoio.dir/build.make:238: modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o] Error 1
```

**Comment by kekePower** [comment]
I get the same message when trying to build against ffmpeg 5.0 on Mageia Cauldron.

Tried to add https://github.com/opencv/opencv/commit/76aff8478883858f0e46746044348ebb16dc3c67 and https://github.com/opencv/opencv/commit/eb067fee5532cd3271f9299c1e69054ded7df71b but it still fails to build.

Build log: https://pkgsubmit.mageia.org/uploads/failure/cauldron/core/release/20220329105355.kekepower.duvel.2754366/log/opencv-4.5.5-4.mga9/build.i586.0.20220329110225.log

**Comment by kekePower** [comment]
This PR fixed the build issue with ffmpeg 5.0: https://github.com/opencv/opencv/pull/21754
